src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -179,13 +214,24 @@ public bool OnDocumentRemoved(DocumentId id)
             {
                 // remove active file state for removed document
                 var removed = false;
-                if (_activeFileStates.TryRemove(id, out _))
+                if (_activeFileStates.ContainsKey(id))
                 {
+                    lock (_lock)
+                    {
+                        _activeFileStates = _activeFileStates.Remove(id);
+                    }
```

Comment:
racey as well.  two things coudl get in here, but it seems like only one actually removed.  again might have been an issue before.